ee/query-service/app/api/license.go

```diff
@@ -88,6 +88,62 @@ func (ah *APIHandler) applyLicense(w http.ResponseWriter, r *http.Request) {
 	ah.Respond(w, license)
 }
 
+func (ah *APIHandler) listLicensesV3(w http.ResponseWriter, r *http.Request) {
+	licenses, apiError := ah.LM().GetLicensesV3(r.Context())
+
+	if apiError != nil {
+		RespondError(w, apiError, nil)
+		return
+	}
+
+	ah.Respond(w, licenses)
+}
+
+// this function is called by zeus when inserting licenses in the query-service
+func (ah *APIHandler) applyLicenseV3(w http.ResponseWriter, r *http.Request) {
+	var license model.LicenseV3
+
+	if err := json.NewDecoder(r.Body).Decode(&license); err != nil {
+		RespondError(w, model.BadRequest(err), nil)
+		return
+	}
```

Comment:
You don't need to explicity decode the entire license as per `model.LicenseV3`.

You can just do the following:
- Decode into a `map[string]any`
- Check for the existence of `map[id] and map[key]`

You should not care about the rest of the information as they can change and will change as per the upstream.

Reply:
done